jsonpickle/util.py

```diff
@@ -43,10 +43,19 @@ def is_type(obj):
     >>> is_type(Klass)
     True
     """
+    # use "isinstance" and not "is" to allow for metaclasses
     if PY3:
-        return type(obj) is type
+        return isinstance(obj, type)
     else:
-        return type(obj) is type or type(obj) is types.ClassType
+        return isinstance(obj, (type, types.ClassType))
+
+
+def has_method(obj, name, *args, **kwargs):
+    try:
+        getattr(obj, name)(*args, **kwargs)
+        return True
+    except:
+        return False
```

Comment:
Your point is absolutely valid, calling a method to check if it's callable is a very cheap way of doing it :) Basically, what we we want to check is that if the attribute points to a bound method, we want the object to be an instance and not a type (which is exactly why it was failing with enum).

Now that I think about it, you can probably get away with something like this:
```python
if getattr(obj, attr).im_self is not None
```